### PR TITLE
cache compensated timer for measuring performance metric

### PR DESCRIPTION
**Pull request by suda-morris**
Play around with git filter-repo, I managed to keep the git history. 

This ccomp_timer was previously located in the esp-idf's tools folder. We found it can be useful for other components' performance tests, so decide to push it to the component registry.

### CODE REVIEW

**Comment by igrr** [comment]
@suda-morris do you think we could squash some of the commits here? A lot of these seem to be "change something unrelated to ccomp_timer and oh by the way also something about ccomp_timer". Perhaps we can keep the original commits from Renz for attribution and then squash most of the remaining ones? (Can keep commit IDs as part of the squashed commit message, just in case...)

**Comment by suda-morris** [comment]
I squashed the origin commits, this time the commit history looks clean. @igrr  @tore-espressif

**Review by tore-espressif — APPROVED** [review]
@suda-morris formal aspects LGTM

File: ccomp_timer/README.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Cache Compensated Timer
+
+[![Component Registry](https://components.espressif.com/components/espressif/ccomp_timer/badge.svg)](https://components.espressif.com/components/espressif/ccomp_timer)
+
+The **Cache Compensated Timer** is a timer that tries to account for instruction and data stall cycles caused by cache misses. It is useful for measuring the time spent in a function or a block of code.
+
+On Xtensa targets (e.g. ESP32), the timer is built on top ot the debug module's performance monitor counter.
+
+On RISC-V targets, this driver is fall-back to the CPU's cycle counter, which actually **doesn't** account for the cache miss.
```
Comment (ESP-Marius) [inline ccomp_timer/README.md:9]:
```suggestion
Due to hardware limitations, on RISC-V targets this driver falls back to using the CPU's cycle counter, which actually **doesn't** account for the cache misses. To achieve a measurement that is independent of cache misses you could place the code is to be measured into IRAM.
```

**Review by ESP-Marius — COMMENTED** [review]
Moving it out LGTM!

**Pull request merged**